### PR TITLE
Update Hydra base so that it works with the latest sqlalchemy. THis i…

### PR DESCRIPTION
**Pull request by knoxsp**
…nvolves removing 'fetchall' commands, udpating the permission query, and the permission relationships, and removing some redundant code

### CODE REVIEW

File: hydra_base/lib/objects.py [hunk 1]
```diff
@@ -26,6 +26,8 @@
 
 from .HydraTypes.Registry import HydraObjectFactory
 
+from sqlalchemy.engine.row import Row
```
Comment (pmslavin) [inline hydra_base/lib/objects.py:29]: This import is unused.

File: hydra_base/util/permissions.py [hunk 2]
```diff
@@ -40,9 +40,13 @@ def check_perm(user_id, permission_code):
...
     try:
-        res = db.DBSession.query(User).join(RoleUser, RoleUser.user_id==User.id).\
-            join(Perm, Perm.id==perm.id).\
-            join(RolePerm, RolePerm.perm_id==Perm.id).filter(User.id==user_id).one()
+        #get all the roles where the specified user has the specified permission
+        qry = db.DBSession.query(RoleUser.role_id)\
+            .join(RolePerm, RolePerm.role_id == RoleUser.role_id)\
+            .filter(RolePerm.perm_id == perm.id)\
+            .filter(RoleUser.user_id == user_id)
+        print (qry.statement)
```
Comment (pmslavin) [inline hydra_base/util/permissions.py:48]: Does this `print` end up in logs or was it just for testing?

**Review by pmslavin — COMMENTED** [review]
All fine: one unused import and one question about a print statement.

File: hydra_base/lib/objects.py (hunk 1 above)
Comment (knoxsp) [reply hydra_base/lib/objects.py]: removed

File: hydra_base/util/permissions.py (hunk 2 above)
Comment (knoxsp) [reply hydra_base/util/permissions.py]: removed

**Pull request merged**